manager/controlapi/service.go

```diff
@@ -327,7 +327,9 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 			return nil
 		}
 		// temporary disable network update
-		if request.Spec != nil && !reflect.DeepEqual(request.Spec.Networks, service.Spec.Networks) {
+		if request.Spec != nil &&
+			(!reflect.DeepEqual(request.Spec.Task.Networks, service.Spec.Task.Networks) ||
+				!reflect.DeepEqual(request.Spec.Networks, service.Spec.Networks)) {
```

Comment:
Maybe we should make an exception for transitioning from `request.Spec.Networks` to `request.Spec.Task.Networks`, if the contents of `Networks` doesn't change.